packages/components/addon/components/hds/table/tr.js

```diff
@@ -0,0 +1,25 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Component from '@glimmer/component';
+import { assert } from '@ember/debug';
+
+export default class HdsTableTrComponent extends Component {
+  /**
+   * @param selectionKey
+   * @type {string}
+   * @default undefined
+   */
+  get selectionKey() {
+    if (this.args.isSelectable && this.args.selectionScope === 'row') {
+      assert(
+        `@selectionKey must be defined when @isSelectable is true`,
```

Comment:
@KristinLBradley we probably need to make the message more meaningful providing more context (eg. it's not clear that this refers to a table row).

<img width="1791" alt="modified" src="https://github.com/hashicorp/design-system/assets/686239/4d481c0a-0147-487f-a952-73f44573254b">

Reply:
Pushed up clarified wording.